engine/StoryState.js

```diff
@@ -729,30 +742,31 @@ export class StoryState{
 
 		copy.outputStream.push.apply(copy.outputStream, this._outputStream);
 		this.OutputStreamDirty();
-		
+
 		copy._currentChoices.push.apply(copy._currentChoices, this._currentChoices);
 
 		if (this.hasError) {
 			copy.currentErrors = [];
 			copy.currentErrors.push.apply(copy.currentErrors, this.currentErrors);
 		}
 
+		if (this.hasWarning) {
+			copy.currentWarnings = [];
+			copy.currentWarnings.push.apply(copy.currentWarnings, this.currentWarnings);
```

Comment:
Thanks for sticking to the existing code style, but boy does that make me doubt my own sanity. A simple `copy. currentWarnings = this.currentWarnings.slice()` would have the same effect as far as I can tell?